test/metabase/api/collection_test.clj

```diff
@@ -77,24 +77,40 @@
         (mt/with-temp* [Collection [collection-1 {:name "Collection 1"}]
                         Collection [collection-2 {:name "Collection 2"}]]
           (perms/grant-collection-read-permissions! (group/all-users) collection-1)
+          (testing "Sanity check: Rasta's personal collection should exist"
+            (is (db/exists? Collection :personal_owner_id (mt/user->id :rasta))))
+          (testing "Sanity check: Rasta should have read perms for her personal Collection"
+            (mt/with-test-user :rasta
+              (is (models/can-read? (db/select-one Collection :personal_owner_id (mt/user->id :rasta))))))
           (is (= ["Our analytics"
                   "Collection 1"
                   "Rasta Toucan's Personal Collection"]
-                 (map :name (mt/user-http-request :rasta :get 200 "collection")))))))
-
-    (testing "check that we don't see collections if they're archived"
-      (mt/with-temp* [Collection [collection-1 {:name "Archived Collection", :archived true}]
-                      Collection [collection-2 {:name "Regular Collection"}]]
-        (is (= ["Our analytics"
-                "Rasta Toucan's Personal Collection"
-                "Regular Collection"]
-               (map :name (mt/user-http-request :rasta :get 200 "collection"))))))
+                 (->> (mt/user-http-request :rasta :get 200 "collection")
+                      (filter (fn [{collection-name :name}]
+                                (or (#{"Our analytics" "Collection 1" "Collection 2"} collection-name)
+                                    (str/includes? collection-name "Personal Collection"))))
+                      (map :name)))))))
+
+    (mt/with-temp* [Collection [collection-1 {:name "Archived Collection", :archived true}]
+                    Collection [collection-2 {:name "Regular Collection"}]]
+      (letfn [(remove-other-collections [collections]
+                (filter (fn [{collection-name :name}]
+                          (or (#{"Our analytics" "Archived Collection" "Regular Collection"} collection-name)
+                              (str/includes? collection-name "Personal Collection")))
+                        collections))]
```

Comment:
If some other test didn't clean up after itself, this code (and changes below) makes sure tests don't fail because random collections show up in the results